test/spec/dataviews/category-dataview/categories-collection.spec.js

```diff
@@ -0,0 +1,43 @@
+var Backbone = require('backbone');
+var CategoriesCollection = require('../../../../src/dataviews/category-dataview/categories-collection');
+
+describe('categories-collection', function () {
+  var aggregationModel;
+  var collection;
+
+  beforeEach(function () {
+    aggregationModel = new Backbone.Model({
+      aggregation: 'count'
+    });
+
+    spyOn(CategoriesCollection.prototype, 'filterNull').and.callThrough();
+    spyOn(CategoriesCollection.prototype, 'reset').and.callThrough();
+
+    collection = new CategoriesCollection(null, {
+      aggregationModel: aggregationModel
+    });
+  });
+
+  it('reset should filter null when reset', function () {
+    collection.reset([{
+      name: 'foo',
+      value: 1
+    }, {
+      name: 'bar',
+      value: 10
+    }, {
+      name: 'wadus',
+      value: null
+    }]);
+
+    collection.reset.calls.reset();
+    expect(CategoriesCollection.prototype.filterNull).toHaveBeenCalled();
```

Comment:
Instead of using spies for this test... why not just test that the collection contains the elements that we are expecting so that the test doesn't know about the internals of the code? eg:

```
expect(collection.pluck('name').toEqual([ 'foo', 'bar', 'wadus' ]);
expect(collection.pluck('value').toEqual([ 1, 10, null ]);
```

or

```
expect(collection.toJSON()).toEqual([{
  name: 'foo',
  value: 1
}, {
  name: 'bar',
  value: 10
}, {
  name: 'wadus',
  value: null
}]);
```